TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs

```diff
@@ -450,7 +450,12 @@ public VehicleRestrictionsTool(TrafficManagerTool mainTool)
                 return false;
             }
 
-            Vector3 center = segment.m_bounds.center;
+            Vector3 center = segment.m_middlePosition;
+            float terrainY = TerrainManager.instance.SampleDetailHeightSmooth(center);
+            if (terrainY > center.y) {
```

Comment:
I don't understand what is this line for?
this is how it looks like 
![image](https://user-images.githubusercontent.com/26344691/128352729-eca106e6-273f-45d4-a0d2-8ca8eb79f357.png)
and this is how it looks like when I comment out the if statement.
![image](https://user-images.githubusercontent.com/26344691/128352867-f2a7a5f3-5de7-47cb-b9b3-e2efb7e9511b.png)
Isn't the second one better?

Reply:
If anything I think that orange overlay should come down.

What do you mean by sunken anyway? you mean like this(the if statement makes no difference in here)?
![image](https://user-images.githubusercontent.com/26344691/128353060-f1dba457-1aef-49d7-981b-a8edcd067136.png)

Reply:
Try vehicle restrictions on the dam 😉

Reply:
but don't you think in trying to fix the dam you might have made the normal use case worse?

Reply:
the dam seems fine with this condition commented out.
![image](https://user-images.githubusercontent.com/26344691/129496269-fe37849c-6a38-4d88-9608-ad9f2f20e55e.png)
So I still don't understand what this if condition does or how sunken segment is relevant to dam.

Reply:
Changed 😉

Reply:
cool it makes more sense now :)